docs/extensions.rst

```diff
@@ -213,3 +213,32 @@ Keys:
 
 The ASGI server will only send the trailing headers in case the client has sent the
 ``TE: trailers`` header in the request.
+
+Debug
+-----
+
+The debug extension allows a way to send debug information from the ASGI application to
+an entity acting as an ASGI server. This extension is not meant to be used in production,
+but only for testing purposes, and ASGI servers should not implement it.
+
+The entity acting as ASGI server will provide ``http.response.debug`` in the extensions
+part of the scope::
+
+    "scope": {
+        ...
+        "extensions": {
+            "http.response.debug": {},
+        },
+    }
+
+The ASGI framework can send debug information by sending a message with the following
+keys. This message can be sent at any time (and multiple times), even before the *Response
+Start* message.
```

Comment:
I think this "any time" may complicate things... Starlette sends it before the *Response Start*, should I be strict and comply on what Starlette does here?

Reply:
Yeah, let's be a bit more specific because it can't hurt, and say it precedes response start. There's only very few scenarios where you'd want it later.

Reply:
Which scenarios do you see "later"?

Reply:
All I could think of was things like "how many sql queries we did while streaming a response out". I was tempted to say we should always make it the _last_ message, but that then doesn't let you test endpoints that stream indefinitely.

Reply:
I think that can already be done by the HTTP trailers extension, if someone wants to do it. I guess...

Reply:
Changed.